ksql-engine/src/test/java/io/confluent/ksql/services/FakeKafkaTopicClient.java

```diff
@@ -69,6 +72,26 @@ private TopicDescription getDescription() {
               .collect(Collectors.toList());
       return new TopicDescription(topicName, false, partitionInfoList);
     }
+
+    @Override
+    public boolean equals(final Object o) {
```

Comment:
Why the introduction of these methods? (Where are they being used?)

Reply:
Sneaky! I check `hasEntry` which in turn uses equals.
```
    assertThat(topicClient.createdTopics(),
        hasEntry(
            "expectedName",
            new FakeTopic("expectedName", 10, (short) 10, TopicCleanupPolicy.DELETE)));
```

Reply:
Ah I see! I thought the methods were being added to `FakeKafkaTopicClient`, not `FakeTopic`. That makes a lot more sense.